tsweb/request_id.go

```diff
@@ -0,0 +1,62 @@
+// Copyright (c) Tailscale Inc & AUTHORS
+// SPDX-License-Identifier: BSD-3-Clause
+
+package tsweb
+
+import (
+	"context"
+	"net/http"
+)
+
+// RequestID is an opaque identifier for a HTTP request, used to correlate
+// user-visible errors with backend server logs. The RequestID is typically
+// threaded through an HTTP Middleware (WithRequestID) and then can be
+// extracted by HTTP Handlers to include in their logs.
+//
+// RequestID is an opaque identifier for a HTTP request, used to correlate
+// user-visible errors with backend server logs. If present in the context, the
+// RequestID will be printed alongside the message text and logged in the
+// AccessLogRecord.
+//
+// There is no particular format required for a RequestID, but ideally it should
+// be obvious to an end-user that it is something to record for support
+// purposes. One possible example for a RequestID format is:
+// REQ-{server identifier}-{timestamp}-{random hex string}.
+type RequestID string
+
+// RequestIDHeader is a custom HTTP header that the WithRequestID middleware
+// uses to determine whether to re-use a given request ID from the client
+// or generate a new one.
+const RequestIDHeader = "X_TAILSCALE_REQUEST_ID"
```

Comment:
For an HTTP header you'll probably want hyphens instead of underscores. Also, you can save Go some extra work canonicalizing the name by doing so here: `X-Tailscale-Request-Id` would be the canonical spelling.

Reply:
100%